src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -889,59 +889,31 @@ private AttributeListSyntax ParseAttributeDeclaration()
                 ? _syntaxFactory.AttributeTargetSpecifier(ConvertToKeyword(this.EatToken()), this.EatToken(SyntaxKind.ColonToken))
                 : null;
 
-            var attributes = _pool.AllocateSeparated<AttributeSyntax>();
-            this.ParseAttributes(attributes);
+            var attributes = this.ParseCommaSeparatedSyntaxList(
+                ref openBracket,
+                SyntaxKind.CloseBracketToken,
+                static @this => @this.IsPossibleAttribute(),
+                static @this => @this.ParseAttribute(),
+                skipBadAttributeListTokens,
```

Comment:
have confirmed in the emitted IL that this reference to the static-local-function (or static-method) only allocates once, tehn caches the delegate.

Reply:
Reviewers may want to jump to the end of this file first to see the docs for this method to understand what's going on when they see these calls (in case it's not self-explanatory).